internal/frame/server.go

```diff
@@ -59,13 +59,25 @@ func NewServer(r io.Reader, w io.Writer) *Server {
 // there is an IO error or an unhandled error is received from the Handler.
 //
 // This blocks until the server is stopped using Stop.
-func (s *Server) Serve(h Handler) error {
+func (s *Server) Serve(h Handler) (retErr error) {
 	if s.running.Swap(true) {
 		return fmt.Errorf("server is already running")
 	}
 
-	defer s.r.Close()
-	defer s.w.Close()
+	defer func() {
+		if err := s.r.Close(); err != nil && retErr == nil {
+			// TODO: by returning this error, internal/plugin/flag_test.go fails
+			// we should be able to handle this error appropriately
```

Comment:
Also, can you let this fail in Travis. I'd like to see how this fails so that
we can fix it.